src/phoenix/server/api/mutations/dataset_mutations.py

```diff
@@ -265,6 +266,88 @@ async def add_examples_to_dataset(
             )
         )
 
+    @strawberry.mutation
+    async def delete_dataset_examples(
+        self, info: Info[Context, None], input: DeleteDatasetExamplesInput
+    ) -> DatasetMutationPayload:
+        example_db_ids = [
+            from_global_id_with_expected_type(global_id, models.DatasetExample.__name__)
+            for global_id in input.example_ids
+        ]
+        # Guard against empty input
+        if not example_db_ids:
+            raise ValueError("Must provide examples to delete")
+        dataset_version_description = (
+            input.dataset_version_description
+            if isinstance(input.dataset_version_description, str)
+            else None
+        )
+        dataset_version_metadata = input.dataset_version_metadata or {}
+        async with info.context.db() as session:
+            # Check if the examples are from a single dataset
+            dataset_ids = (
+                (
+                    await session.execute(
+                        select(models.DatasetExample.dataset_id)
+                        .where(models.DatasetExample.id.in_(example_db_ids))
+                        .distinct()
+                    )
+                )
+                .scalars()
+                .all()
+            )
+            if len(dataset_ids) > 1:
+                raise ValueError("Examples must be from the same dataset")
+            elif not dataset_ids:
+                raise ValueError("Examples not found")
+
+            # Get the dataset and raise an error if it doesn't exist
+            dataset_db_id = dataset_ids[0]
+
+            dataset_version_rowid = await session.scalar(
+                insert(models.DatasetVersion)
+                .values(
+                    dataset_id=dataset_db_id,
+                    description=dataset_version_description,
+                    metadata_=dataset_version_metadata,
```

Comment:
might want to set the same `create_at` for the version and revisions

Reply:
good call